### ISSUE TITLE
Axolotl?

### ISSUE THREAD

**Issue by jprichardson** [tradle/zlorp#1]
As stated https://github.com/tradle/tim#zlorp:

> Zlorp is just the core chat module. It uses OTR for secure sessions (later will add support for Axolotl, used in TextSecure and Pond). 

Any plans / timelines to get Axolotl working?

**Comment by mvayngrib** [comment]
@jprichardson planned, but no timeline at the moment, too busy. Do you want to give it a try? It's mostly this [chunk](https://github.com/tradle/zlorp/blob/master/lib/peer.js#L141-L207) in peer.js. I was looking at [this implementation](https://github.com/alexeykudinkin/axolotl.js), @alexeykudinkin does good work. Another candidate was [this one](https://github.com/joebandenburg/libaxolotl-javascript) but [this](https://github.com/joebandenburg/libaxolotl-javascript/issues/3#issuecomment-70423842) has me a bit wary. Your thoughts?

**Comment by jprichardson** [comment]
> @jprichardson planned, but no timeline at the moment, too busy. Do you want to give it a try?

I really want to put a secure chat / data communication system in http://www.exodus.io/ but it's not a priority for me just yet. But planning for when it will be :)

> It's mostly this chunk in peer.js. I was looking at this implementation, @alexeykudinkin does good work. Another candidate was this one but this has me a bit wary. Your thoughts?

I really like that https://github.com/alexeykudinkin/axolotl.js uses elliptic (albeit an outdated version), as I'd like to the community converge to one ecdsa library and elliptic is the best. I glanced over https://github.com/joebandenburg/libaxolotl-javascript and related libraries, but didn't really like the author's coding style (no offense to author meant). .. but I figured with one of these libraries, I'll have to dive in and help contribute.

Another option, would be to compile https://github.com/WhisperSystems/libaxolotl-c with Emscripten. I'm not a fan of doing this, but it may provide the best security / compatibility with the protocol. Or, another idea, use this same method, but only use it to verify against the work in https://github.com/alexeykudinkin/axolotl.js with the intent to ultimately use axolotl.js.

**Comment by mvayngrib** [comment]
@jprichardson good idea, didn't think of testing axolotl.js with compiled libaxolotl-c. exodus looks beautiful, will keep an eye on it! Where's the repo?

**Comment by alexeykudinkin** [comment]
@jprichardson there were plans of ultimately aligning protocol versions with actual one of the TS protocol to make it fully-compatible. Though it isn't top-priority one.
